pkg/operator/resources/job/batchapi/metrics.go

```diff
@@ -17,216 +17,139 @@ limitations under the License.
 package batchapi
 
 import (
+	"context"
+	"fmt"
 	"time"
 
-	"github.com/aws/aws-sdk-go/aws"
-	"github.com/aws/aws-sdk-go/service/cloudwatch"
 	"github.com/cortexlabs/cortex/pkg/lib/errors"
 	"github.com/cortexlabs/cortex/pkg/lib/parallel"
-	"github.com/cortexlabs/cortex/pkg/lib/slices"
 	"github.com/cortexlabs/cortex/pkg/operator/config"
 	"github.com/cortexlabs/cortex/pkg/types/metrics"
 	"github.com/cortexlabs/cortex/pkg/types/spec"
+	promapi "github.com/prometheus/client_golang/api"
+	promv1 "github.com/prometheus/client_golang/api/prometheus/v1"
+	"github.com/prometheus/common/model"
 )
 
-// Find data retention period at https://aws.amazon.com/cloudwatch/faqs/
-
-func getCompletedBatchMetrics(jobKey spec.JobKey, startTime time.Time, endTime time.Time) (*metrics.BatchMetrics, error) {
-	batchMetrics := metrics.BatchMetrics{}
+const (
+	_metricsRequestTimeoutSeconds = 10
+)
 
-	if time.Now().Sub(endTime) < 2*time.Hour {
-		return getRealTimeBatchMetrics(jobKey)
+func getBatchMetrics(jobKey spec.JobKey) (metrics.BatchMetrics, error) {
+	client, err := promapi.NewClient(promapi.Config{
+		Address: config.Cluster.PrometheusURL,
+	})
+	if err != nil {
+		return metrics.BatchMetrics{}, err
 	}
 
-	minimumEndTime := time.Now()
-	if time.Now().Sub(endTime) > 2*time.Minute {
-		minimumEndTime = endTime.Add(2 * time.Minute)
-	}
+	promAPIv1 := promv1.NewAPI(client)
```

Comment:
nit: this can be moved to a global client (e.g. k8s client is in the config package)